commit 6d8e0f84f89f99c87608569e374d52cf248978ee
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Mon Jul 18 16:04:05 2022 +0200

    wifi: mac80211: mlme: set sta.mlo to mlo state
    
    At this point, we've already changed link_id to be zero for
    a non-MLO connection, so use the 'mlo' variable rather than
    link ID to determine the MLO status of the station.
    
    Fixes: bd363ee53302 ("wifi: mac80211: mlme: set sta.mlo correctly")
    Fixes: 81151ce462e5 ("wifi: mac80211: support MLO authentication/association with one link")
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 2d389e74c953..1ced0a4b428e 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -6393,7 +6393,7 @@ static int ieee80211_prep_connection(struct ieee80211_sub_if_data *sdata,
 	}
 
 	if (!have_sta) {
-		if (link_id >= 0)
+		if (mlo)
 			new_sta = sta_info_alloc_with_link(sdata, ap_mld_addr,
 							   link_id, cbss->bssid,
 							   GFP_KERNEL);
@@ -6405,7 +6405,7 @@ static int ieee80211_prep_connection(struct ieee80211_sub_if_data *sdata,
 			goto out_err;
 		}
 
-		new_sta->sta.mlo = link_id >= 0;
+		new_sta->sta.mlo = mlo;
 	}
 
 	/*